packages/api/src/util/helper.ts

```diff
@@ -27,3 +31,67 @@ export function extractEthereumSignature(signatures: EthereumSignature[]): { r:
 
   return { r: rPart, s: sPart, v: vPart };
 }
+
+// Helper function to deposit claim that will return when claim is successfully deposited
+export async function awaitDepositClaim(
+  api: Api,
+  depositTxHash: H256 | string | Uint8Array,
+  claim: { tokenAddress: string; amount: string; beneficiary: string },
+  signer: KeyringPair
+): Promise<ClaimDeposited> {
+  const nonce = await api.rpc.system.accountNextIndex(signer.address);
+  return new Promise((resolve, reject) => {
+    api.tx.erc20Peg
+      .depositClaim(depositTxHash, claim)
+      .signAndSend(signer, { nonce }, ({ status, events }) => {
+        let eventClaimId;
+        if (status.isInBlock) {
+          for (const {
+            event: { method, section, data },
+          } of events) {
+            if (section === 'erc20Peg' && method === 'Erc20Claim') {
+              eventClaimId = data[0];
+              break;
+            } else if (section === 'system' && method === 'ExtrinsicFailed') {
+              reject('Claim already notarized');
+            }
+          }
+        }
+        if (eventClaimId) {
+          api.query.system
+            .events((events) => {
```

Comment:
why query events again?

Reply:
to search for Erc20Deposit/Erc20DepositFail event... and then resolve the promise cause the deposit has already completed/failed

Reply:
When we check events for the first time.. those events are only generated via sign and send.. It will only get events related to depositClaim extrinsic... the second time, events are generic from everywhere..